docs/src/reference/compilation.rst

```diff
@@ -429,6 +429,64 @@ running session.  Please check `Sage documentation
 You can tailor the behavior of the Cython compiler by specifying the
 directives below.
 
+.. _compiling_notebook:
+
+Compiling with a Jupyter Notebook
+=================================
+
+It's possible to compile code in a notebook cell with Cython.
+For this you need to load the Cython magic::
+
+    %load_ext Cython
+
+Then you can define a Cython cell by writing ``%%cython`` on top of it.
+Like this::
+
+    %%cython
+
+    cdef int a = 0
+    for i in range(10):
+        a += i
+    print(a)
+
+Note that each cell will be compiled individually. So if you use a package in a Cython
```

Comment:
I think this would be a bit clearer if it said "compiled into a separate extension module". Because that is the main reason why globals are not shared across cells in both directions. However, the global (Python) names of the cell are then loaded into the global namespace of the notebook, which I think is worth mentioning here.